components/DatePicker/DatePicker.tsx

```diff
@@ -1,33 +1,37 @@
-import { useState } from "react";
+import { useEffect } from "react";
 
 import Image from "next/image";
 
 import classNames from "classnames/bind";
 import dayjs, { Dayjs } from "dayjs";
 
 import { DAYS } from "@/constants/calendarConstants";
+import useScroll from "@/hooks/useScroll";
+import { useDateStore } from "@/store/useDateStore";
 import { calculateMonthDates, divideWeek } from "@/utils/calculateCalendarDates";
 
 import styles from "./datePicker.module.scss";
+import MonthNavButton from "./MonthNavButton";
 
 const cn = classNames.bind(styles);
 
 export interface DatePickerProps {
-  selectedDay: Dayjs;
-  setSelectedDay: (date: Dayjs) => void;
   hasNavigation?: boolean;
 }
 
-export default function DatePicker({
-  selectedDay,
-  setSelectedDay,
-  hasNavigation = true,
-}: DatePickerProps) {
-  const [viewDate, setViewDate] = useState(selectedDay);
+export default function DatePicker({ hasNavigation = true }: DatePickerProps) {
+  const { focusDate, viewDate, setFocusDate, setViewDate } = useDateStore();
+
+  useEffect(() => {
+    setFocusDate(dayjs());
+    setViewDate(dayjs());
+  }, [setFocusDate, setViewDate]);
 
   const handleDateClick = (day: Dayjs) => {
-    setSelectedDay(day);
-    setViewDate(day);
+    if (hasNavigation) {
+      setFocusDate(day);
+      setViewDate(day);
+    }
```

Comment:
```suggestion
    if (!hasNavigation) {
      return;
    }
    setFocusDate(day);
    setViewDate(day);
```
제 개인적인 취향은 early return 사용하거나 사용처에서 ` onClick={() => hasNavigation && handleDateClick(day)}` And 연산자 거는건 별로 일까요??

Reply:
사용처에서 거는거 저 저번에 해봤는데 그렇게하면 뭔가 클릭했을때 깜빡이는 게 잠깐 보인다해야하나..? 그래서 일단 이부분은 두도록 하겠습니다..